### ISSUE TITLE
Warning for Automated Task and Image Cleanup Configuration

### ISSUE THREAD

**Issue by murphyslaw** [aws/amazon-ecs-agent#557]
We receive a warning from the ecs-agent

```
2016-10-12T14:13:34Z [WARN] Invalid format for "ECS_NUM_IMAGES_DELETE_PER_CYCLE", expected an integer. err strconv.ParseInt: parsing "": invalid syntax
```

We are using the the ECS optimized AMI and the following ecs-agent version

```
{
    "Cluster": "<cluster>",
    "ContainerInstanceArn": "<arn>",
    "Version": "Amazon ECS Agent - v1.13.0 (aebcbca)"
}
```

For testing purposes our `/etc/ecs/ecs.config` currently looks like

```
ECS_CLUSTER=<cluster>
ECS_DISABLE_IMAGE_CLEANUP=false
ECS_IMAGE_CLEANUP_INTERVAL=10m
ECS_IMAGE_MINIMUM_CLEANUP_AGE=1h
ECS_NUM_IMAGES_DELETE_PER_CYCLE=10
```

We pushed a couple of images, but none of them got cleaned up. Currently this feature is not working for us.

After removing the standard `ecs-agent` container and starting it manually with the command from the README

```
docker run --name ecs-agent \
    --detach=true \
    --restart=on-failure:10 -d \
    --volume=/var/run/docker.sock:/var/run/docker.sock \
    --volume=/var/log/ecs:/log \
    --volume=/var/lib/ecs/data:/data \
    --net=host \
    --env-file=/etc/ecs/ecs.config \
    --env=ECS_LOGFILE=/log/ecs-agent.log \
    --env=ECS_DATADIR=/data/ \
    --env=ECS_ENABLE_TASK_IAM_ROLE=true \
    --env=ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST=true \
    amazon/amazon-ecs-agent:latest
```

the warning disappeared from the log and it deleted an unreferenced image. Still, we have at the moment roughly 100 docker images in the particular ECS repository and we set the `ECS_NUM_IMAGES_DELETE_PER_CYCLE` to `10`, but the cleanup task only deleted a single image and left all others untouched.

**Comment by murphyslaw** [comment]
After looking at the Go code and running `docker inspect ecs-agent` we saw that the standard agent has `ECS_AGENT_CONFIG_FILE_PATH` configured to `/etc/ecs/ecs.config.json`, so we created this file and tried to play around with possible values.

The configuration file we ended up with is

```
{
  "Cluster": "<cluster>",
  "ImageCleanupDisabled": false,
  "NumImagesToDeletePerCycle": 10,
  "ImageCleanupInterval": 600000000000,
  "MinimumImageDeletionAge": 3600000000000
}
```

We looked up the JSON keys from the `Config` struct in https://github.com/aws/amazon-ecs-agent/blob/master/agent/config/config.go. The code also does not support to use time duration strings like `"10m"` or `"1h"` in the JSON file, so we used integer values that will be parsed as `ns` values.

The warning still remains even when using a configuration file. It seems, because the environment variables are always read in the beginning and then potentially overwritten by the configuration values from the configuration file. Only for the `ECS_NUM_IMAGES_DELETE_PER_CYCLE` there is no default value configured in https://github.com/aws/amazon-ecs-agent/blob/master/agent/config/config.go#L317, so it will always create the warning in all cases where it is not set as an environment variable.
